### PR TITLE
feat(docs): use `mkdocs-click` for documenting EEST CLIs

### PR DESCRIPTION
**Pull request by danceratopz**
## 🗒️ Description
This PR ensures that our CLI documentation is single source by using:
1. [`click`](https://click.palletsprojects.com)'s built-in help system to document CLIs (for terminal output), and,
2. [`mkdocs-click`](https://github.com/mkdocs/mkdocs-click), that generates html documentation from the click docstrings.

It also updates the `evm_bytes` CLI docs to have cleaner `click` docs and use `mkdocs-click` for its html docs. This fixes up the previous poor formatting of help strings in the terminal.

Online docs example output:
image

## 🔗 Related Issues
None.

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Review by winsvega — APPROVED** [review]

File: src/cli/evm_bytes.py [hunk 1]
```diff
@@ -148,94 +148,87 @@ def process_evm_bytes_string(evm_bytes_hex_string: str, assembly: bool = False)
...
-@cli.command()
+@evm_bytes.command(short_help="Convert a binary file to Python Opcodes or assembly.")
 @assembly_option
-@click.argument("binary_file_path", type=click.File("rb"))
-def binary_file(binary_file_path, assembly: bool):
+@click.argument("binary_file", type=click.File("rb"))
+def binary_file(binary_file, assembly: bool):
     """
-    Convert the given EVM bytes binary file.
+    Convert the BINARY_FILE containing EVM bytes to Python Opcodes or assembly.
```
Comment (raxhvl) [inline src/cli/evm_bytes.py:210]:
```suggestion
    Convert the `binary_file` containing EVM bytecode to Python Opcodes or assembly.
```

File: src/cli/evm_bytes.py [hunk 2]
```diff
@@ -148,94 +148,87 @@ def process_evm_bytes_string(evm_bytes_hex_string: str, assembly: bool = False)
...
+
+    BINARY_FILE is a binary file containing EVM bytes, use `-` to read from stdin.
```
Comment (raxhvl) [inline src/cli/evm_bytes.py:212]:
```suggestion
    `binary_file` is a binary file containing EVM bytecode, use `-` to read from stdin.
```

File: src/cli/evm_bytes.py [hunk 3]
```diff
@@ -148,94 +148,87 @@ def process_evm_bytes_string(evm_bytes_hex_string: str, assembly: bool = False)
...
 
 
-@cli.command()
+@evm_bytes.command(short_help="Convert a hex string to Python Opcodes or assembly.")
 @assembly_option
 @click.argument("hex_string")
 def hex_string(hex_string: str, assembly: bool):
     """
-    Process a hex string representing EVM bytes and convert it into EEST's Python opcodes.
+    Convert the HEX_STRING representing EVM bytes to EEST Python Opcodes.
```
Comment (raxhvl) [inline src/cli/evm_bytes.py:170]:
```suggestion
    Convert the `hex_string` representing EVM bytecode to EEST Python Opcodes.
```
nit for consistency.

File: src/cli/evm_bytes.py (hunk 2 above)
Comment (danceratopz) [reply src/cli/evm_bytes.py:212]: Using uppercase is suggested in the click docs. It'd be nice if it did this for you, but it doesn't. 

Writing as uppercase means you have a match with the "Usage" string in the terminal.

The blue box shows what happens with backticks
image

Comment (danceratopz) [reply src/cli/evm_bytes.py:212]: But tbh, "bytes" matches well with the name of the CLI `evm_bytes` :smile:

Comment (danceratopz) [reply src/cli/evm_bytes.py:212]: > Using uppercase is suggested in the click docs. It'd be nice if it did this for you, but it doesn't.

https://click.palletsprojects.com/en/stable/documentation/#documenting-arguments

Comment (raxhvl) [reply src/cli/evm_bytes.py:212]: > But tbh, "bytes" matches well with the name of the CLI evm_bytes 😄
agreed.

**Pull request merged**